source/MaterialXCore/Util.cpp

```diff
@@ -175,4 +177,21 @@ string parentNamePath(const string& namePath)
     return EMPTY_STRING;
 }
 
+string getBaseCompoundName(const string& nodeName, const string& typeNames, const string& version, const string& namespaceString) {
```

Comment:
Seems weird to me that it's assumed that the nodeName, typeName, and version include the '_' but the namespaceString does not. Maybe none of them should include the '_' and this method should add it. Thoughts?

Reply:
I agree, it is a bit of a legacy that the typeNames have a "_", I will fix this.